code/game/area/areas.dm

```diff
@@ -291,16 +291,31 @@ var/list/mob/living/forced_ambiance_list = new
 		for(var/obj/machinery/door/window/temp_windoor in src)
 			temp_windoor.open()
 
-/area/proc/has_gravity()
+/area/has_gravity()
 	return has_gravity
 
 /area/space/has_gravity()
 	return 0
 
-/proc/has_gravity(atom/AT, turf/T)
-	if(!T)
-		T = get_turf(AT)
-	var/area/A = get_area(T)
+/atom/proc/has_gravity()
+	var/area/A = get_area(src)
+	if(A && A.has_gravity())
+		return 1
+	return 0
+
+/mob/has_gravity()
+	if(istype(loc, /turf/space))
```

Comment:
Can't you just do `loc.has_gravity()` to cover space and area?

Reply:
we have a reference of the area so usually it's cheaper to just use the area.

Reply:
No you don't, you get the area reference several lines below here.